arbor/fvm_lowered_cell_impl.hpp

```diff
@@ -164,6 +164,15 @@ void fvm_lowered_cell_impl<Backend>::reset() {
 
     update_ion_state();
 
+    // Call initialize again
```

Comment:
This comment needs to be expanded to explain why we are double-initializing the ion species and mechanisms. See the comment below that explains why `threshold_watcher_` is reset at that point.

Reply:
I will update the comment.
However, there's another issue in ticket #896 (Scenario 2) that is not addressed in this PR yet. Initially we were going to not allow that scenario to occur, but after re-discussing it with Sam we agreed it's better to add support for it. So I will update this PR in the next few days.